src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -4308,132 +4306,118 @@ private void EmitInlineConditionalNode(AssociativeNode node, ref ProtoCore.Type
-                    EmitDynamicLanguageBlockNode(langblockT, bExprTrue, ref inferedType, ref trueBlockId, graphNode, ProtoCore.CompilerDefinitions.SubCompilePass.None);
-                    List<GraphNode> trueBodyNodes = core.InlineConditionalBodyGraphNodes.Pop();
-
-                    // Append dependent nodes of the inline conditional 
-                    foreach (GraphNode gnode in trueBodyNodes)
-                        foreach (GraphNode dNode in gnode.dependentList)
-                            graphNode.PushDependent(dNode);
-
-                    core.AssocNode = null;
-                    DynamicBlockNode dynBlockT = new DynamicBlockNode(trueBlockId);
-
-                    // False condition language block
-                    BinaryExpressionNode bExprFalse = AstFactory.BuildReturnStatement(inlineConditionalNode.FalseExpression);
+            };
 
-                    LanguageBlockNode langblockF = new LanguageBlockNode();
-                    int falseBlockId = Constants.kInvalidIndex;
-                    langblockF.codeblock.Language = ProtoCore.Language.Associative;
-                    core.AssocNode = bExprFalse;
-                    core.InlineConditionalBodyGraphNodes.Push(new List<GraphNode>());
-                    EmitDynamicLanguageBlockNode(langblockF, bExprFalse, ref inferedType, ref falseBlockId, graphNode, ProtoCore.CompilerDefinitions.SubCompilePass.None);
-                    List<GraphNode> falseBodyNodes = core.InlineConditionalBodyGraphNodes.Pop();
+            // As SSA conversion is enabled, we have got the values of
+            // true and false branch, so it isn't necessary to create 
+            // language blocks.
+            if (core.Options.GenerateSSA)
+            {
+                inlineCall.FormalArguments.Add(inlineConditionalNode.ConditionExpression);
+                inlineCall.FormalArguments.Add(inlineConditionalNode.TrueExpression);
+                inlineCall.FormalArguments.Add(inlineConditionalNode.FalseExpression);
+            }
+            else
+            {
+                // True condition language block
+                BinaryExpressionNode bExprTrue = AstFactory.BuildReturnStatement(inlineConditionalNode.TrueExpression);
+
+                LanguageBlockNode langblockT = new LanguageBlockNode();
+                int trueBlockId = Constants.kInvalidIndex;
+                langblockT.codeblock.Language = ProtoCore.Language.Associative;
+                core.AssocNode = bExprTrue;
+                core.InlineConditionalBodyGraphNodes.Push(new List<GraphNode>());
+                EmitDynamicLanguageBlockNode(langblockT, bExprTrue, ref inferedType, ref trueBlockId, graphNode,
+                    ProtoCore.CompilerDefinitions.SubCompilePass.None);
+                List<GraphNode> trueBodyNodes = core.InlineConditionalBodyGraphNodes.Pop();
+
+                // Append dependent nodes of the inline conditional 
+                foreach (GraphNode gnode in trueBodyNodes)
```

Comment:
find this style with no braces kind of difficult to read / prone to future errors - what do you think?

Reply:
I actually find the requirement to have the extra braces unnecessary and creates code that is more unreadable (mostly due to adding unnecessary code and code lines), but the bracing wars is a long running issue across multiple companies, I really have no dog in this race. That being said, these changes were made by Aparajit, I can change them however you would like.